client/state/posts/reducer.js

```diff
@@ -282,6 +281,20 @@ export function edits( state = {}, action ) {
 				[ action.siteId ]: omit( state[ action.siteId ], action.postId || '' )
 			} );
 
+		case POST_SAVE_SUCCESS:
+			if ( ! state.hasOwnProperty( action.siteId ) || ! action.savedPost || action.postId ) {
+				break;
+			}
+			const { siteId, savedPost } = action;
```

Comment:
Minor, not worth another round of reviews: Could move destructuring above the condition to make the condition slightly shorter:

```js
const { siteId, savedPost, postId } = action;
if ( ! state.hasOwnProperty( siteId ) || ! savedPost || postId ) {
```